### ISSUE TITLE
letsencrypt stops apache, then fails

### ISSUE THREAD

**Issue by singpolyma** [letsencrypt/letsencrypt#1638]
Relevant log except:

```
2015-11-30 21:18:48,589:INFO:letsencrypt.reporter:Reporting to user: The following 'urn:acme:error:connection' errors were reported by the server
:

Domains: roundcube.singpolyma.net
Error: The server could not connect to the client for DV
2015-11-30 21:18:48,614:INFO:letsencrypt.auth_handler:Cleaning up challenges
2015-11-30 21:18:48,782:ERROR:letsencrypt_apache.configurator:Apache Reload Failed!
Reloading web server: apache2 failed!
Apache2 is not running ... (warning).
```

After which apache is not running.  It was running before I ran letsencrypt (running from master, latest) and starting it again works fine.  It seems that letsencrypt is stopping it and then complaining that it is stopped...

**Comment by bmw** [comment]
@singpolyma, thank you very much for reporting this. If possible, I would love to see the full log.

**Comment by singpolyma** [comment]
@bmw not sure what all is in that log... is there a way I can submit privately?

**Comment by bmw** [comment]
Would you feel comfortable e-mailing it to me? My e-mail is bmw@eff.org.

**Comment by singpolyma** [comment]
sent

**Comment by bmw** [comment]
Received. Thanks again.

This issue brought my attention to a different problem that's in `master` but not yet in PyPI. Thanks for reporting this before I pushed the changes!

The problem you experienced is odd. In terms of modifications to Apache, the following occurred:

1. Some changes were made to your Apache configuration.
2. `/etc/init.d/apache2 reload` was executed and returned a zero status. It was assumed that the reload was successful.
3. `boulder`, the server side of Let's Encrypt, tried to perform domain validation and failed to connect to your server.
4. The changes in 1 were reverted.
5. `/etc/init.d/apache2 reload` was executed and returned a non-zero status saying Apache was not running.

This means that the reload in step 2 actually failed despite it's status of zero. Per Apache documentation, this is possible but I've never seen it in practice.
